backend/EkoFunkcje/EkoFunkcje/Features/Interventions/AddInterventionFunction.cs

```diff
@@ -41,17 +42,24 @@ public AddInterventionFunction(IAddressConverter addressConverter)
                 string json = JsonConvert.SerializeObject(errorList);
                 return new BadRequestObjectResult(json);
             }
-            Address convertedGeoAddress = new Address();
+
+            Address convertedGeoAddress;
             try
             {
                 // retry should be added
-                convertedGeoAddress = await _addressConverter.ConvertToGeoAddress(intervention.Address);
+                convertedGeoAddress = await _addressConverter.ConvertToGeoAddress(intervention.City, intervention.Street, intervention.StreetNumber);
+            }
+            catch (BaseException e)
+            {
+                log.Log(e.LogLevel, e, e.Message);
+                return new BadRequestObjectResult(e.Message);
             }
             catch (Exception e)
             {
-                log.LogError(e, "error");
-                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+                log.LogError(e, "Podany adres jest niepoprawny");
```

Comment:
Zastanawiam sie czy nie powinnismy tutaj jednak zostawic 500. Obecnie jak leci exception który powinien zwrócić 500 (niespodziewany) to dostajemy 400 z "Podany adres jest niepoprawny". Prawdziwe exceptiony dotyczace adresu wyłapujemy we wcześniejszym bloku z tego co rozumiem.

Reply:
Zmienione znowu na 500